wolfssl/wolfcrypt/wc_port.h

```diff
@@ -82,6 +82,16 @@
     #endif
 #endif /* !WC_MAYBE_UNUSED */
 
+#ifndef WC_DEPRECATED
+    #if !defined(WOLFSSL_ZEPHYR) && \
+            ((defined(__GNUC__) && (__GNUC__ >= 4)) || defined(__clang__) || \
+            defined(__IAR_SYSTEMS_ICC__))
+        #define WC_DEPRECATED __attribute__((deprecated))
+    #else
```

Comment:
Do we want to support this with MSVC?

Reply:
There's lots of other targets that can express the `deprecated` attribute one way or another -- we'll add them as we go, but I focused on the targets I knew well.  I didn't want to let the perfect be the enemy of the good here.